frontend/public/components/search.tsx

```diff
@@ -0,0 +1,264 @@
+  const clearLabelFilter = () => {
+    setLabelFilter([]);
+    setSelector(selectorFromString(''));
+    setQueryArgument('q', '');
+  };
+
+  const clearAll = () => {
+    setSelectedItems([]);
+    setLabelFilter([]);
+    setNameFilter([]);
+    setSelector(selectorFromString(''));
+    setQueryArgument('kind', '');
+    setQueryArgument('q', '');
+    setQueryArgument('name', '');
+  };
+
+  const isKindExpanded = (kindView: string) => {
+    return collaspedState.includes(kindView);
+  };
+
+  const toggleKindExpanded = (kindView: string) => {
+    const newCollasped = new Set([...collaspedState]);
+    newCollasped.has(kindView) ? newCollasped.delete(kindView) : newCollasped.add(kindView);
+    setCollaspedState([...newCollasped]);
+  };
+
+  const updateNameFilter = (value: string) => {
+    setNameFilter([...nameFilter, value]);
+    setQueryArgument('name', [...nameFilter, value].map(encodeURIComponent).join(','));
+  };
+
+  const updateLabelFilter = (value: string) => {
+    if (requirementFromString(value) !== undefined) {
+      setLabelFilter([...labelFilter, value]);
+      setSelector(selectorFromString([...labelFilter, value].join(',')));
+      setQueryArgument('q', [...labelFilter, value].map(encodeURIComponent).join(','));
+    }
+  };
+
+  const updateSearchFilter = (type: string, value: string) => {
+    type === 'label' ? updateLabelFilter(value) : updateNameFilter(value);
+  };
+
+  const removeNameFilter = (value: string) => {
+    const newNames = nameFilter.filter((keepItem: string) => keepItem !== value);
+    setNameFilter(newNames);
+    setQueryArgument('name', newNames.map(encodeURIComponent).join(','));
+  };
+
+  const removeLabelFilter = (value: string) => {
+    const newLabels = labelFilter.filter((keepItem: string) => keepItem !== value);
+    setLabelFilter(newLabels);
+    setQueryArgument('q', newLabels.map(encodeURIComponent).join(','));
+  };
+
+  const getToggleText = (item: string) => {
+    const kindName = item.split('~').pop();
+    if (selectedItems.length === 1) {
+      return kindName;
+    }
+    return isKindExpanded(item) ? `Hide ${kindName}` : `Show ${kindName}`;
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>Search</title>
+      </Helmet>
+      <PageHeading detail={true} title="Search">
+        <div className="co-search-group">
+          <ResourceListDropdown
+            selected={selectedItems}
+            onChange={updateSelectedItems}
+            className="co-search-group__resource"
+          />
+          <SearchFilterDropdown onChange={updateSearchFilter} />
+        </div>
+        <div className="form-group">
+          <ChipGroup withToolbar defaultIsOpen={false}>
+            <ChipGroupToolbarItem key="resources-category" categoryName="Resource">
+              {selectedItems.map((chip) => (
+                <Chip key={chip} onClick={() => updateSelectedItems(chip)}>
+                  <ResourceIcon kind={chip} />
+                  {chip.split('~').pop()}
```

Comment:
Use `kindForReference`.

Note that this could get confusing if selecting the same kind name in different API groups. There'd be no way to tell them apart :/